File: Tree/BT_level_order_traversal.cpp
/*
Tree : visited array 대신 parent array && depth array 사용
레벨 순회(Level-order Traversal) : 레벨 순서데로 순회 
*/
#include <bits/stdc++.h>
using namespace std;

int lc[9] = {0,2,4,6,0,0,0,0,0};
int rc[9] = {0,3,5,7,0,8,0,0,0};

void BFS(){
    queue<int> Q;
    Q.push(1);  // root = 1;
    while(!Q.empty()){
        int cur = Q.front(); Q.pop();
        cout << cur << ' ';
        if(lc[cur]) Q.push(lc[cur]);
        if(rc[cur]) Q.push(rc[cur]);
    }
}

int main(){
    BFS();
}

File: Tree/BT_level_order_traversal.py
lc = [0, 2, 4, 6, 0, 0, 0, 0, 0]
rc = [0, 3, 5, 7, 0, 8, 0, 0, 0]

from collections import deque

def BFS(root):
    Q = deque()
    Q.append(root)
    while (Q):
        curr = Q.popleft()
        print(curr, end = " ")
        if (lc[curr]):
            Q.append(lc[curr])
        if (rc[curr]):
            Q.append(rc[curr])
            
BFS(1)
    
